### DISCUSSION TITLE
Help Creating Custom Claim

### DISCUSSION THREAD

**Discussion by Kryszn0** [zitadel/zitadel#7153] [Q&A]
I've referenced the following guide ( https://zitadel.com/blog/custom-claims ), but can't seem to be able to figure it out. I am trying to pass user roles as groups in NGINX using the following:

`auth_request_set $groups $upstream_http_x_auth_request_grants;
proxy_set_header REMOTE-GROUPS $groups;`

Note: the only modification I've made to the 'flatRoles' example from Zitadel's site is rename the claim from 'my:zitadel:grants' to 'grants'

**Comment by hifabienne** [comment]
Hei @K-J-VV 
Can you add some more information?
What does not work? Do you get an error? How do your claims look like?

**Reply by Kryszn0** [reply]
Thanks for the response, I was able to find an answer in another [thread](https://github.com/argoproj/argo-cd/discussions/11855#discussioncomment-5586777).

I am not a developer, just an advanced user. Loving Zitadel so far, it is very high quality and powerful. Just trying to learn how to use it from the standpoint of a non-developer, so things like creating custom claims is a little beyond me
